scripts/mbedtls_framework/unittest_config_checks.py

```diff
@@ -0,0 +1,195 @@
+"""Test the configuration checks that reject some bad compile-time configs.
+
+This tests the output of ``generate_config_checks.py``.
+This can also let us verify what we enforce in the manually written
+checks in ``<PROJECT>_check_config.h``.
+"""
+
+## Copyright The Mbed TLS Contributors
+## SPDX-License-Identifier: Apache-2.0 OR GPL-2.0-or-later
+
+import os
+import subprocess
+import sys
+import unittest
+from typing import List, Optional, Pattern, Union
+
+
+class TestConfigChecks(unittest.TestCase):
+    """Unit tests for checks performed via ``<PROJECT>_config.c``.
+
+    This can test the code generated by `config_checks_generator`,
+    as well as manually written checks in `check_config.h`.
+    """
+
+    # Set this to the path to the source file containing the config checks.
+    PROJECT_CONFIG_C = None #type: Optional[str]
+
+    # Project-specific include directories (in addition to /include)
+    PROJECT_SPECIFIC_INCLUDE_DIRECTORIES = [] #type: List[str]
+
+    # Increase the length of strings that assertion failures are willing to
+    # print. This is useful for failures where the preprocessor has a lot
+    # to say.
+    maxDiff = 9999
+
+    def setUp(self) -> None:
+        self.cpp_output = None #type: Optional[str]
+
+    def tearDown(self) -> None:
+        """Log the preprocessor output to a file, if available and desired.
+
+        This is intended for debugging. It only happens if the environment
+        variable UNITTEST_CONFIG_CHECKS_DEBUG is non-empty.
+        """
+        if os.getenv('UNITTEST_CONFIG_CHECKS_DEBUG'):
+            # We set self.cpp_output to the preprocessor output before
+            # asserting, and set it to None if all the assertions pass.
+            if self.cpp_output is not None:
+                basename = os.path.splitext(os.path.basename(sys.argv[0]))[0]
+                filename = f'{basename}.{self._testMethodName}.out.txt'
+                with open(filename, 'w') as out:
+                    out.write(self.cpp_output)
+
+    def user_config_file_name(self, variant: str) -> str:
+        """Construct a unique temporary file name for a user config header."""
+        name = os.path.splitext(os.path.basename(sys.argv[0]))[0]
+        pid = str(os.getpid())
+        oid = str(id(self))
+        return f'tmp-user_config_{variant}-{name}-{pid}-{oid}.h'
+
+    def write_user_config(self, variant: str, content: Optional[str]) -> Optional[str]:
+        """Write a user configuration file with the given content.
+
+        If content is None, ensure the file does not exist.
```

Comment:
```suggestion
        If content is None, remove the file (if it exists).
```

Isn't this clearer?

Reply:
“Ensure the file does not exist” says what we do: it's the purpose of this aspect of the function. “Remove the file (if it exists)” says how we do it. The point of documentation is to explain what we do and why: if you want to know how, the code tells you. So I think “ensure the file does not exist” is better in the documentation.